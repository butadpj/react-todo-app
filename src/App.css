.app {
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-flow: column;
}

header {
  position: fixed;
  top: 10%;
  font-size: 2em;
  color: var(--light);
  font-weight: 800;
  z-index: -1;
  animation: fadeOut 1s ease-out 2.5s forwards;
}

/* CLASS UTILITIES START */
.btn {
  border: none;
  padding: 1em 1.5em;
  color: black;
  cursor: pointer;
  transition: 0.3s ease-out;
}

.btn:hover {
  filter: brightness(120%);
  box-shadow: 0 3px 12px rgba(0, 0, 0, 0.20);
}

.btn--primary {
  background: var(--primary);
  color: var(--light);
}

.btn--danger {
  background: var(--danger);
  color: var(--light);
}

.btn--success {
  background: var(--success);
  color: var(--light);
}

.btn--transparent {
  background: none;
  color: var(--light);
}

.completed {
  text-decoration: line-through;
}
/* CLASS UTILITIES END */

/* TODO LIST START */
.todo-list {
  width: 100%;
  max-width: 40rem;
  list-style: none;
  overflow: auto;
  padding-bottom: 10rem;
  color: var(--light);
  font-size: 1.5em;
}

.todo-list li {
  display: flex;
  flex-flow: column;
  justify-content: center;
  align-items: center;
  padding: 1em;
  border-bottom: 1px solid var(--primary);
}

.todo-list li:hover {
  background: var(--primary);
}

.todo__buttons {
  margin-top: 1rem;
}

.delete-btn {
  background: transparent;
  color: white;
}

.delete-btn:hover {
  background: var(--danger);
  color: var(--light);
}

/* BUTTONS HOVER */
.todo-list li:hover .done-btn{
  background: var(--success);
  color: var(--light);
}

.todo-list li:hover .undo-btn{
  background: var(--light);
  color: var(--dark);
}
/* TODO LIST END */

/* TODO NAV START */
.todo-nav {
  position: fixed;
  left: 0;
  bottom: 67px;
  width: 100%;
  max-width: 40rem;
}

.nav__icon {
  align-self: flex-end;
}

.nav__buttons {
  display: flex;
  flex-flow: row wrap;
  justify-content: space-between;
  align-items: center;
}

.nav__buttons > * {
  flex-basis: 100%;
}
/* TODO NAV END */

/* TODO FORM START */
.todo-form {
  position: fixed;
  bottom: 0;
  width: 100%;
  max-width: 40rem;
  display: flex;
}

.form__input {
  width: 100%;
  font-size: 1.2rem;
  padding-left: 1em;
  border: none;
}

.form__input:focus {
  outline: none;
  border: 1px solid var(--dark);
}

.add-btn {
  font-size: 1.2rem;
}
/* TODO FORM END */

@media only screen and (min-width: 40rem) {
  .todo-list li {
    flex-flow: row;
    justify-content: space-between;
  }

  .todo__buttons {
    display: flex;
    margin: 0 0 0 1rem;
  }

  .nav__buttons > * {
    flex-basis: auto;
  }
}

/* ANIMATIONS */
@keyframes fadeOut {
  0% {
    opacity: 1;
  }
  100% {
    opacity: 0;
  }
}