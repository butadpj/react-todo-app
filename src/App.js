import React, { useState } from 'react';
import { Header, TodoForm, TodoList } from './components';
import './App.css';

const App = () => {
  const [todoInput, setTodoInput] = useState('');
  const [todos, setTodos] = useState([]);

  const handleInputChange = (event) => {
    setTodoInput(event.target.value);
  }

  const handleInputSubmit = (event) => {
    event.preventDefault();

    if (todoInput) {
      setTodoInput('');    
      setTodos([
        ...todos, // Copy of previous object in array
        {
          id: new Date().valueOf(), 
          name: todoInput,
          isDone: false,
        }, 
      ]);
    }
  }

  const handleToggleIsDone = (selectedTodo) => {
    let updatedTodo = {...selectedTodo, isDone: !selectedTodo.isDone};
    setTodos(todos.map((todo) => 
      // Only return the updated todo (finishedTodo) 
      // and just copy the rest
      todo.id === updatedTodo.id ? updatedTodo : todo
    ));
  }

  const handleDeleteTodo = (selectedTodoId) => {
    setTodos(todos.filter((todo) => selectedTodoId !== todo.id))
  }

  return (
    <div className="app">
      <Header>Todo App</Header>
      <TodoList 
        todos={todos} 
        toggleIsDone={handleToggleIsDone}
        deleteTodo={handleDeleteTodo}
      />
      <TodoForm 
        value={todoInput} 
        onChange={handleInputChange}
        onSubmit={handleInputSubmit}
      />
    </div>
  );
}

export default App;
