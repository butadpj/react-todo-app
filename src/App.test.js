import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';

describe("<App />", () => {
  it("renders without crashing", () => {
    shallow(<App />);
  });

  it("UI doesn't change unexpectedly", () => {
    const wrapper = mount(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});