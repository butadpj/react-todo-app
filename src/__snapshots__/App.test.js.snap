// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> UI doesn't change unexpectedly 1`] = `
<App>
  <div
    className="app"
  >
    <Header>
      <header>
        Todo App
      </header>
    </Header>
    <TodoList
      deleteTodo={[Function]}
      todos={Array []}
      toggleIsDone={[Function]}
    >
      <ul
        className="todo__list"
      >
        <EmptyTodoMessage>
          <p
            className="empty-todo-message"
            style={
              Object {
                "padding": "0 1.5rem",
                "textAlign": "center",
              }
            }
          >
            Seems quiet here... Add your first todo.
          </p>
        </EmptyTodoMessage>
      </ul>
    </TodoList>
    <TodoForm
      onChange={[Function]}
      onSubmit={[Function]}
      value=""
    >
      <form
        className="todo-form"
        onSubmit={[Function]}
      >
        <input
          autoFocus={true}
          className="todo__input"
          onChange={[Function]}
          placeholder="Add your new todo"
          type="text"
          value=""
        />
        <button
          className="btn btn--success add-btn"
          type="submit"
        >
          Add
        </button>
      </form>
    </TodoForm>
  </div>
</App>
`;
